mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -203,6 +203,33 @@ static llvm::Value *createAtomicRMW(llvm::IRBuilderBase &builder,
                                   {ptr->getType(), val->getType()}, {ptr, val});
 }
 
+// Create a call to GenISA_dpas for matrix multiply-add.
+static llvm::Value *
+createGenISADPAS(GENX::MatrixDPASOp op, llvm::IRBuilderBase &builder,
+                 LLVM::ModuleTranslation &moduleTranslation) {
+  llvm::Module *module = builder.GetInsertBlock()->getModule();
+  auto opTypes = op->getOperandTypes();
+  llvm::Function *fn = llvm::GenISAIntrinsic::getDeclaration(
+      module, llvm::GenISAIntrinsic::GenISA_dpas,
+      {moduleTranslation.convertType(op->getResultTypes()[0]),
+       moduleTranslation.convertType(opTypes[0]),
+       moduleTranslation.convertType(opTypes[1]),
+       moduleTranslation.convertType(opTypes[2])});
+  // FIXME: Attribute mismatch - Attribute 'noundef' does not apply to
```

Comment:
@pengtu Is the attribute mismatch due to llvm mismatch?

Reply:
This is very likely due to the LLVM IR mismatch we discussed. The FIXME IMO can be addressed after peng work on IGC is done.

Reply:
Yes, the workaround is at https://github.com/pengtu/llvm-project/blob/genx/mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp#L228